lib/config/validation.ts

```diff
@@ -634,6 +634,26 @@ export async function validateConfig(
                   message: `Invalid \`${currentPath}.${key}.${res}\` configuration: value is not a string`,
                 });
               }
+            } else if (key === 'encrypted') {
+              const res = validatePlainObject(val);
+              if (res !== true) {
+                errors.push({
+                  topic: 'Configuration Error',
+                  message: `Invalid \`${currentPath}.${key}.${res}\` configuration: value is not a string`,
+                });
+              }
+
+              const privateKey =
+                configType === 'global'
+                  ? (config.privateKey ?? config.privateKeyOld)
+                  : (GlobalConfig.get('privateKey') ??
+                    GlobalConfig.get('privateKeyOld'));
+              if (!is.nonEmptyString(privateKey)) {
+                errors.push({
+                  topic: 'Configuration Error',
+                  message: `No privateKey found in hosted config. \`encrypted\` object cannot be used without a privateKey.`,
```

Comment:
This error needs to be different when MEND_HOSTED="true". See discussion

Reply:
can we use `config.strings` for it instead of adding a mend specific env? so self hosted users can define custom message.